### PR TITLE
Level up

### LINKED ISSUE #30: US: NPC has chosen level

### PR DESCRIPTION
**Pull request by LHJE**
### Code Highlights:
 * Allows users to determine the level of their NPC, and the HP and stats will increase.
 * The stats the NPC's "choose" to increase at the levels they're allowed to are chosen at random
 * Adds ability for Users to delete their account
### Where should the reviewer start?
 * `app/poros/stats.rb`
### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.
### Issues:
* Addresses #30
* Closes #30
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [X] Rubocop Violations: 88.  The long lines and methods are still caused by the specific nature of the different classes.  I'll refactor as I can going forward.
- [x] SimpleCov Coverage: 98%: Because of the random nature of `npc_service_spec.rb`, `equipment.rb`, `npc.rb`, `npc_facade.rb`, and `core_stats.rb`, there will always be a few lines of code that aren't being hit when testing.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,6 +20,22 @@ def create
...
+        NpcModelWeapon.where(npc_model_id: id).destroy_all
+        NpcModelSpell.where(npc_model_id: id).destroy_all
+        UserNpcModel.where(npc_model_id: id).destroy_all
+      end
+      NpcModel.where(id: user_npc_model.npc_model_id).destroy_all
+    end
+    user.destroy
+    redirect_to root_path
+  end
+
```
Comment (LHJE) [inline app/controllers/users_controller.rb:38]: So Users don't have to keep their account if they don't want

File: app/poros/core_stats.rb [hunk 2]
```diff
@@ -60,6 +61,12 @@ def distribute_buffs(ancestry_buffs)
     end
   end
 
+  def distribute_leveled_stats(leveled_stats)
+    leveled_stats.each do |attr, score|
+      @stats[:scores][attr.to_s.to_sym] += score
+    end
+  end
```
Comment (LHJE) [inline app/poros/core_stats.rb:68]: This is how we're handling leveling up the stats.

File: app/poros/stats.rb [hunk 3]
```diff
@@ -3,17 +3,123 @@ class Stats
...
+    end
+  end
+
+  def level_up(level, class_name)
+    if class_name == 'fighter'
+      level_up_fighter(level)
+    else
+      level_up_non_fighter(level)
+    end
+  end
```
Comment (LHJE) [inline app/poros/stats.rb:37]: Ooooo boy, here's where the trouble starts.  Fighter's ability scores go up more often than anyone else's, so what lies below is a lot, and will need to be refactored.

File: app/views/shared/_npc_show.html.erb [hunk 4]
```diff
@@ -28,7 +29,13 @@
...
       <% @weapons.each do |weapon|%>
         <li>Name: <%= weapon.name %></li>
-        <li>Damage: <%= weapon.damage %></li>
+        <% if weapon.damage == 'No Damage'%>
+          <li>Damage: <%= weapon.damage %> </li>
+        <% elsif weapon.classification.include?("Ranged") || weapon.properties.include?("Finesse") && @npc.str_attack <= @npc.dex_attack %>
+          <li>Damage: <%= weapon.damage %> + <%= @npc.dex_attack %></li>
+        <% else %>
+          <li>Damage: <%= weapon.damage %> + <%= @npc.str_attack %></li>
+        <% end %>
```
Comment (LHJE) [inline app/views/shared/_npc_show.html.erb:38]: This'll show the best type of attack bonus for your build!

**Review by cecile1981 — APPROVED** [review]

**Pull request merged**